librz/type/base.c

```diff
@@ -188,9 +188,28 @@ RZ_API RZ_OWN RzBaseType *rz_type_base_type_new(RzBaseTypeKind kind) {
  * \param typedb Type Database instance
  * \param type RzBaseType to save
  */
-RZ_API void rz_type_db_save_base_type(const RzTypeDB *typedb, const RzBaseType *type) {
-	rz_return_if_fail(typedb && type && type->name);
-	ht_pp_insert(typedb->types, type->name, (void *)type);
+RZ_API bool rz_type_db_save_base_type(const RzTypeDB *typedb, RzBaseType *type) {
+	rz_return_val_if_fail(typedb && type && type->name, false);
+	if (!ht_pp_insert(typedb->types, type->name, (void *)type)) {
+		rz_type_base_type_free(type);
+		return false;
+	}
+	return true;
+}
+
+/**
+ * \brief Saves RzBaseType into the Types DB
```

Comment:
`Updates the base type, frees the old one, and frees the type if update failed` or something along these lines